src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -57,16 +63,28 @@ private String retrieveItemAttribute(String response, String flag) {
         return response.trim().substring(startOfItemAttribute, endOfItemAttribute);
     }
 
-    private int retrieveEventBudget(String response) {
+    private void initializeDateTime(String taskDeadline) throws DukeException {
+        try {
+            itemDateTime = Parser.convertDateTime(taskDeadline);
+            if (itemDateTime.isBefore(LocalDateTime.now())) {
+                throw new DukeException("Unfortunately, we cannot travel back in time. Please "
+                        + "enter a valid date and time in the format 'dd-MM-yyyy HHmm'. ");
+            }
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Please use this format for the deadline 'dd-MM-yyyy HHmm'. ");
```

Comment:
This is a good exception that was thrown as this looks at not only the formatting of the date and time, but whether the date and time has already passed.